waku/v2/node/keepalive.go

```diff
@@ -59,13 +59,22 @@ func (w *WakuNode) startKeepAlive(ctx context.Context, randomPeersPingDuration t
 
 		select {
 		case <-allPeersTickerC:
-			relayPeersSet := make(map[peer.ID]struct{})
-			for _, t := range w.Relay().Topics() {
-				for _, p := range w.Relay().PubSub().ListPeers(t) {
-					relayPeersSet[p] = struct{}{}
+			if w.opts.enableRelay {
+				relayPeersSet := make(map[peer.ID]struct{})
+				for _, t := range w.Relay().Topics() {
+					for _, p := range w.Relay().PubSub().ListPeers(t) {
+						relayPeersSet[p] = struct{}{}
+					}
 				}
+				peersToPing = append(peersToPing, maps.Keys(relayPeersSet)...)
 			}
-			peersToPing = maps.Keys(relayPeersSet)
+
+			// We ping all filter nodes
+			filterPeersSet := make(map[peer.ID]struct{})
```

Comment:
Since we are pinging filter peers in shorter interval these are not required. 
Also, shouldn't this be in else condition of `w.opts.enableRelay` as relay and filter will never be enabled together?